other/max_sum_sliding_window.py

```diff
@@ -0,0 +1,47 @@
+"""
+Given an array of integer elements and an integer 'k', we are required to find the
+maximum sum of 'k' consecutive elements in the array.
+
+Instead of using a nested for loop, in a Brute force approach we will use a technique
+called 'Window sliding technique' where the nested loops can be converted to a single
+loop to reduce time complexity.
+"""
+import sys
+from typing import List
+
+
+def max_sum(array: List[int], k: int) -> int:
+    """
+    Returns the maximum sum of k consecutive elements
+    >>> arr = [1, 4, 2, 10, 2, 3, 1, 0, 20]
+    >>> k = 4
+    >>> max_sum(arr, k)
+    24
+    >>> k = 10
+    >>> max_sum(arr,k)
+    Traceback (most recent call last):
+        ...
+    ValueError: Invalid Input
+    >>> arr = [1, 4, 2, 10, 2, 13, 1, 0, 2]
+    >>> k = 4
+    >>> max_sum(arr, k)
+    27
+    """
+    if len(array) < k or k < 0:
+        raise ValueError("Invalid Input")
+    max_sum = -sys.maxsize
+    current_sum = sum(array[:k])

```

Comment:
```suggestion
    current_sum = max_sum = sum(array[:k])
```
You can remove `import sys` as well.
Also, change either the variable name or the function name as they are the same.

Reply:
You just added my test :)
Try to do some random testing. Maybe use `unittest` or `pytest` as `doctest` can get quite cumbersome when you want to do large scale testing.

Reply:
Actually, its a more conventional way of doing the same thing. Beginners are told to do just like that, so I think they may find it a bit easier to understand if I leave it unchanged.

Reply:
I am using random testing

Reply:
Try to think yourself: what do you think is more readable and easier to understand?
1. Putting the first value as the reference
2. Putting an arbitrary negative value depending on the system which the program is running on as the reference

I would choose 1 all the time and I think beginners will understand 1 more easily.

And if you're referring to the block `if __name__ == "__main__":` then that's just a demo for your program, it's not actually testing anything.

Reply:
About that first suggestion, most beginner-friendly solutions are written that way. I personally will find that as easier to read and understand. (Maybe its personal preferences). It just makes the code more beginner-friendly (IMO).
Yeah, I am not familiar with `unittest` and `pytest`. Will surely look it up and add it from now on if needed. 👍

Reply:
Also, as `sys.maxsize` is a fix value, what would happen when the maximum sum value in the array is less than that value? That would give us the wrong answer.

Reply:
If you want to use that approach then I recommend using `float('-inf')` otherwise a problem can arise as mentioned in the previous comment.

Reply:
Yep. I made the change